code/modules/xenoarcheaology/machinery/geosample_scanner.dm

```diff
@@ -59,40 +62,54 @@
 	return TRUE
 
 /obj/machinery/radiocarbon_spectrometer/attackby(var/obj/I, var/mob/user)
-	if(scanning)
-		to_chat(user, "<span class='warning'>You can't do that while [src] is scanning!</span>")
-	else
-		if(istype(I, /obj/item/stack/nanopaste))
-			var/choice = alert("What do you want to do with the nanopaste?","Radiometric Scanner","Scan nanopaste","Fix seal integrity")
-			if(choice == "Fix seal integrity")
-				var/obj/item/stack/nanopaste/N = I
-				var/amount_used = min(N.get_amount(), 10 - scanner_seal_integrity / 10)
-				N.use(amount_used)
-				scanner_seal_integrity = round(scanner_seal_integrity + amount_used * 10)
-				return
-		if(istype(I, /obj/item/chems/glass))
-			var/choice = alert("What do you want to do with the container?","Radiometric Scanner","Add coolant","Empty coolant","Scan container")
-			if(choice == "Add coolant")
-				var/obj/item/chems/glass/G = I
-				var/amount_transferred = min(src.reagents.maximum_volume - src.reagents.total_volume, G.reagents.total_volume)
-				G.reagents.trans_to(src, amount_transferred)
-				to_chat(user, "<span class='info'>You empty [amount_transferred]u of coolant into [src].</span>")
-				update_coolant()
-				return
-			else if(choice == "Empty coolant")
-				var/obj/item/chems/glass/G = I
-				var/amount_transferred = min(G.reagents.maximum_volume - G.reagents.total_volume, src.reagents.total_volume)
-				src.reagents.trans_to(G, amount_transferred)
-				to_chat(user, "<span class='info'>You remove [amount_transferred]u of coolant from [src].</span>")
-				update_coolant()
-				return
+	if(istype(I, /obj/item/stack/nanopaste))
+		if(scanning)
+			to_chat(user, SPAN_WARNING("You can't do that while [src] is scanning!"))
+			return
+		var/choice = alert("What do you want to do with the nanopaste?","Radiometric Scanner","Scan nanopaste","Fix seal integrity")
+		if(choice == "Fix seal integrity")
+			var/obj/item/stack/nanopaste/N = I
+			var/amount_used = min(N.get_amount(), 10 - scanner_seal_integrity / 10)
+			N.use(amount_used)
+			scanner_seal_integrity = round(scanner_seal_integrity + amount_used * 10)
+			return TRUE
+	if(istype(I, /obj/item/chems/glass))
+		if(scanning)
+			to_chat(user, SPAN_WARNING("You can't do that while [src] is scanning!"))
+			return
+		var/choice = alert("What do you want to do with the container?","Radiometric Scanner","Add coolant","Empty coolant","Scan container")
+		if(choice == "Add coolant")
```

Comment:
Wouldn't that fit in a refactor more than in fix for unit tests though? Because this thing has many issues, and faulty user input is basically the least one of them?

Reply:
Yeah, its just that, from experience, even the dumbest change can break things, and since that's a fix PR, I'm trying to minimize collateral damage ^^;
I guess I can do it.